### DISCUSSION TITLE
db.statement does not capture SQL

### DISCUSSION THREAD

**Discussion by mitht** [open-telemetry/opentelemetry-java-instrumentation#11492] [Q&A]
The project utilizes com.alibaba.druid.proxy.jdbc.PreparedStatementProxyImpl to enhance the functionality of PreparedStatement. When using the OpenTelemetry agent, the db.statement does not capture the correct SQL. How should I solve this issue? {...db.statement=com.alibaba.druid.proxy.jdbc.PreparedStatementProxyImpl@?}

**Comment by laurit** [comment]
> How should I solve this issue?

I doubt that anybody can give you precise instructions for that without debugging this and figuring out what is wrong. The code for the jdbc instrumentation is in https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/jdbc. Use whatever debugging techniques you know to figure out why it doesn't work.
Alternatively provide a minimal sample application that reproduces your issue so someone else could take a look.

**Reply by mitht** [reply]
<dependency>
            <groupId>com.alibaba</groupId>
            <artifactId>druid</artifactId>
            <version>1.2.16</version>
</dependency>

  datasource:
    driver-class-name: org.postgresql.Driver
    type: com.alibaba.druid.pool.DruidDataSource
    initialSize: 5
    minIdle: 5
    maxActive: 20
    maxWait: 60000
    timeBetweenEvictionRunsMillis: 60000
    minEvictableIdleTimeMillis: 300000
    validationQuery: SELECT 'x'
    testWhileIdle: true
    testOnBorrow: false
    testOnReturn: false
    filters: stat,wall

If the Druid filters: stat, wall are enabled; db.statement=com.alibaba.druid.proxy.jdbc.PreparedStatementProxyImpl@570c696c; unable to capture SQL.
<img width="996" alt="image" src="https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/29425860/b6eba1f7-29ad-48d7-9712-edd9713d8066">

**Reply by laurit** [reply]
If you wish for someone to take a look at this you'll have to provide a minimal application that reproduces the issue.
